modular_splurt/code/datums/traits/neutral.dm

```diff
@@ -623,33 +623,94 @@
 	. = ..()
 
 /datum/quirk/nudist
-	// Mostly derived from masked_mook.
-	// Spawning with a gear harness is preferable, but failed during testing.
 	name = "Nudist"
 	desc = "Wearing most types of clothing unnerves you. Bring a gear harness!"
 	gain_text = span_notice("You feel spiritually connected to your natural form.")
 	lose_text = span_notice("It feels like clothing could fit you comfortably.")
 	medical_record_text = "Patient expresses a psychological need to remain unclothed."
 	value = 0
 	mood_quirk = TRUE
-	processing_quirk = TRUE
+	var/is_nude
 
-/datum/quirk/nudist/on_process()
-	var/mob/living/carbon/human/H = quirk_holder
-	// Checking torso exposure appears to be a robust method.
-	if( ( H.is_chest_exposed() && H.is_groin_exposed() ) )
-		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, QMOOD_NUDIST, /datum/mood_event/nudist_positive)
-	else
-		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, QMOOD_NUDIST, /datum/mood_event/nudist_negative)
+/datum/quirk/nudist/add()
+	// Register signal handlers
+	RegisterSignal(quirk_holder, COMSIG_MOB_UPDATE_GENITALS, .proc/check_outfit)
```

Comment:
shouldn't this signal be unregistered too?

Reply:
It is unregistered in the unregister list, as part of remove.